frontend/src/tests/application-tests.js

```diff
@@ -1,25 +1,268 @@
+    const posting = {
+        name: "CSC209F TA",
+        intro_text: "Testing posting for CSC209F",
+        open_date: "2021/04/01",
+        close_date: "2021/05/01",
+        availability: "open",
+    };
+
+    const postingPosition = {
+        num_positions: 10,
+        hours: 68,
+    };
+
+    const newUser = {
+        roles: ["ta"],
+        utorid: "greenb",
+    };
+
     beforeAll(async () => {
         await databaseSeeder.seed({ apiGET, apiPOST });
+        session = databaseSeeder.seededData.session;
+        applicant = databaseSeeder.seededData.applicant;
+        position = databaseSeeder.seededData.position;
+
+        // get the current active user
+        let resp = await apiGET("/debug/active_user");
+        expect(resp).toHaveStatus("success");
+        defaultUser = resp.payload;
     }, 30000);
 
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
-        it.todo("Get survey.js posting data through public route");
+        it("Get survey.js posting data through public route", async () => {
+            // make sure the user has admin permissions before this post request
+            // make a new posting and update <posting> to include the id of the posting
+            resp = await apiPOST("/admin/postings", {
+                ...posting,
+                session_id: session.id,
+            });
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(postingPropTypes, resp.payload);
+            expect(resp.payload.id).not.toBeNull();
+            Object.assign(posting, resp.payload);
+
+            // make sure the user is the seeded applicant before this get request for the purposes of the next test
+            // create a user for the applicant to make sure they exist before changing the default user to applicant
+            const applicantAsUser = {
+                roles: ["ta"],
+            };
+            applicantAsUser.utorid = applicant.utorid;
+
+            resp = await apiPOST("/debug/users", applicantAsUser);
+            expect(resp).toHaveStatus("success");
+            Object.assign(applicantAsUser, resp.payload);
+
+            // set the default user to the applicant
+            resp = await apiPOST("/debug/active_user", applicantAsUser);
+            expect(resp).toHaveStatus("success");
+
+            // read survey.js posting data
+            resp = await apiGET(`/public/postings/${posting.url_token}`, true);
+            expect(resp).toHaveStatus("success");
+            checkPropTypes(surveyPropTypes, resp.payload);
+            surveyPrefill = resp.payload["prefilled_data"];
+
+            // restore the default user
+            resp = await apiPOST("/debug/active_user", defaultUser);
+            expect(resp).toHaveStatus("success");
+        });
+
+        // broken up into 2 tests: applicant & application(not yet implemented in the backend)
```

Comment:
Reference an issue number. Otherwise it will be very hard to know if this has been resolved.